scrapy/commands/genspider.py

```diff
@@ -69,7 +69,17 @@ def run(self, args, opts):
         try:
             spidercls = self.crawler_process.spider_loader.load(name)
         except KeyError:
-            pass
+            if not self.settings.get('NEWSPIDER_MODULE'):
+                if exists(name + ".py"):
+                    print("%r already exists" % (abspath(name + ".py")))
+                    return
+            else:
+                spiders_module = import_module(self.settings['NEWSPIDER_MODULE'])
+                spiders_dir = dirname(spiders_module.__file__)
+                spiders_dir_abs = abspath(spiders_dir)
+                if exists(join(spiders_dir_abs, name + ".py")):
+                    print("%r already exists" % (join(spiders_dir_abs, (name + ".py"))))
+                    return
```

Comment:
Looking at lines 84-88, I suspect this implementation will fail even if `--force` is used.

Reply:
Hmm...I had discussed the same with @elacuesta https://github.com/scrapy/scrapy/pull/4623#discussion_r439445481 and he said since it was just a matter of renaming the file which the user could do easily it wasn't required. However, I could add it if its needed.

Reply:
Right, sorry about that. I missed the fact that the `--force` option already exists for this command. Given that, it probably makes sense to take the flag into account.

Reply:
No worries, I'll add it then